### PR TITLE
ENH: extracted class DICOMReferencesDialog and simplified/refactored code 

### PR DESCRIPTION
**Pull request by che85**
I thought that might help for improving readability

image

### CODE REVIEW

**Comment by pieper** [comment]
Looks reasonable.  Thanks!

**Comment by jcfr** [comment]
Thanks for the PR.

The docker images have been rebuilt. It means CircleCI is expected to be green.

File: Modules/Scripted/DICOMLib/DICOMWidgets.py [hunk 1]
```diff
@@ -906,11 +847,81 @@ def onNodeAdded(caller, event, calldata):
...
+                "confirm if you would like to load the following referenced datasets."
+
+  def __init__(self, parent, loadables):
+    super(DICOMReferencesDialog, self).__init__(parent)
+    self.loadables = loadables
+    self.checkboxes = dict()
+    self.setup()
+
+  def setup(self):
+    self._setBasicProperties()
```
Comment (jcfr) [inline Modules/Scripted/DICOMLib/DICOMWidgets.py]: If that helps, you create the UI using designer and include it as resources.
For example, see https://github.com/Slicer/Slicer/blob/eefeac9286f48552e8418a11f412b2823a09b407/Modules/Scripted/ExtensionWizard/CMakeLists.txt#L18-L20 that is referencing a ui file loaded dynamically.

It is also possible to install it in location relative the python script and load it from there.

**Comment by che85** [comment]
I think that's ready for merge ;)

**Comment by jcfr** [comment]
Thanks. Integrated in [r25916](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=25916) to [r25919](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=25919)

**Pull request closed without merging**